### PR TITLE
[WIP][SDFAB-638] Create GH Action for building P4C image

### PR DESCRIPTION
**Pull request by EmanueleGallone**
This PR will port the CI, currently on Travis, to GH Actions.

The workflow requires 2 secrets to be specified within this repository:
* DOCKER_USERNAME (DockerHub's username)
* DOCKER_PASSWORD (DockerHub's token)

### CODE REVIEW

**Comment by EmanueleGallone** [comment]
@ccascone should I remove the Travis CI stuff? (Or maybe just edit the `README.md` removing the Travis CI references since it won't be used anymore..?)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Change file permissions
+        run: chmod +x docker_push
+        
+      - name: Run docker_push
+        run: ./docker_push "$DOCKER_PASSWORD" "$DOCKER_USERNAME" "$DOCKER_RUNTIME_IMAGE" "$LOCAL_RUNTIME_IMAGE"
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+          DOCKER_RUNTIME_IMAGE: ${{ steps.before_install.outputs.DOCKER_RUNTIME_IMAGE }}
+          LOCAL_RUNTIME_IMAGE : ${{ steps.before_install.outputs.DOCKER_RUNTIME_IMAGE }}
```
Comment (daniele-moro) [inline .github/workflows/main.yml:89]: As we discussed, we can most probably directly use the GH action do do build and pull

Comment (EmanueleGallone) [reply .github/workflows/main.yml:89]: Yes. I'll try to use https://github.com/marketplace/actions/build-and-push-docker-images

**Comment by EmanueleGallone** [comment]
I can't get the GH actions to work anymore, for testing (maybe related to this https://github.community/t/github-action-workflow-is-executed-for-a-pr-from-a-forked-repo-if-the-pr-changes-the-workflow-file/16059); I'll close this PR and created another one after the refactor of the workflow has been done.

**Comment by EmanueleGallone** [comment]
Nevermind, apparently there are currently some issues with the platform itself: https://www.githubstatus.com/incidents/bdbzpz7qxmbx

**Comment by ccascone** [comment]
@EmanueleGallone yes we can remove the Travis stuff

**Pull request closed without merging**